### PR TITLE
KAFKA-14133: Migrate stateManager mock in StoreChangelogReaderTest to Mockito

### PR DESCRIPTION
**Pull request by clolov**
This pull request takes a similar approach to how TaskManagerTest is being migrated to Mockito mock by mock for easier reviews.

### CODE REVIEW

**Comment by clolov** [comment]
Heya @cadonna! This pull request is aiming to move one of the three remaining classes to Mockito mock by mock. I would be very grateful for a review when you have the time 😊

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 1]
```diff
@@ -194,19 +206,19 @@ public void shouldNotRegisterSameStoreMultipleTimes() {
...
             () -> changelogReader.register(new TopicPartition("ChangelogWithoutStoreMetadata", 0), stateManager));
     }
 
     @Test
     public void shouldSupportUnregisterChangelogBeforeInitialization() {
+        setupStateManagerMock();
         final Map<TaskId, Task> mockTasks = mock(Map.class);
         EasyMock.expect(mockTasks.get(null)).andReturn(mock(Task.class)).anyTimes();
         EasyMock.expect(storeMetadata.offset()).andReturn(9L).anyTimes();
-        EasyMock.expect(stateManager.changelogOffsets()).andReturn(singletonMap(tp, 10L));
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:214]: This (along with other similar expectations) was flagged as unneeded by Mockito

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 2]
```diff
@@ -946,7 +966,6 @@ public void shouldOnlyRestoreStandbyChangelogInUpdateStandbyState() {
         final Map<TaskId, Task> mockTasks = mock(Map.class);
         EasyMock.expect(mockTasks.get(null)).andReturn(mock(Task.class)).anyTimes();
         EasyMock.expect(mockTasks.containsKey(null)).andReturn(true).anyTimes();
-        EasyMock.expect(stateManager.taskType()).andReturn(type);
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:949]: This setup, and the two below, was reported as unnecessary by Mockito

**Review by divijvaidya — APPROVED** [review]

**Comment by divijvaidya** [comment]
There are flaky tests in CI but the test changed in this PR is successful: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14929/2/testReport/org.apache.kafka.streams.processor.internals/StoreChangelogReaderTest/

**Pull request merged**